uniffi_bindgen/src/scaffolding/mod.rs

```diff
@@ -79,12 +79,23 @@ mod filters {
         })
     }
 
-    // Map types to their fully-qualified `FfiConverter` impl.
+    // Map a type to Rust code that specifies the FfiConverter implementation.
+    //
+    // This outputs something like `<TheFfiConverterStruct as FfiConverter>`
     pub fn ffi_converter(type_: &Type) -> Result<String, askama::Error> {
-        Ok(format!(
-            "<{} as uniffi::FfiConverter<crate::UniFfiTag>>",
-            type_rs(type_)?
-        ))
+        Ok(match type_ {
+            Type::External {
+                name,
+                kind: ExternalKind::Interface,
+                ..
+            } => {
+                format!("<::std::sync::Arc<r#{name}> as uniffi::FfiConverter<crate::UniFfiTag>>")
```

Comment:
This is a slight step backwards from the changes I made in #1444 to make this more uniform.  However, I don't think it will get in the way of the goal of simplifying things for the proc-macro code.  In any case, proc-macros don't support external types yet and we can figure this out later.